File: fix_sass.py
import os
import glob

# Buscar todos los style.scss en los bloques de src
base_path = r"c:\Users\UPIC\Local Sites\vpindev\app\public\wp-content\themes\vpinunf\src\blocks"
files = glob.glob(os.path.join(base_path, "**", "style.scss"), recursive=True)

count = 0
for f in files:
    try:
        with open(f, 'r', encoding='utf-8') as file:
            content = file.read()
            
        if '@import "../../scss/base/mixins";' in content:
            content = content.replace('@import "../../scss/base/mixins";', '@use "../../scss/base/mixins" as *;')
            with open(f, 'w', encoding='utf-8') as file:
                file.write(content)
            count += 1
    except Exception as e:
        print(f"Error {f}: {e}")

print(f"Se actualizaron {count} archivos.")

File: fix_sass.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

walkDir("C:\\Users\\UPIC\\Local Sites\\vpindev\\app\\public\\wp-content\\themes\\vpinunf\\src\\blocks", function(filePath) {
  if (filePath.endsWith('.scss')) {
    let data = fs.readFileSync(filePath, 'utf-8');
    if (data.includes('@import "../../scss/base/mixins";')) {
      let result = data.replace('@import "../../scss/base/mixins";', '@use "../../scss/base/mixins" as *;');
      fs.writeFileSync(filePath, result, 'utf8');
      console.log('Fixed:', filePath);
    }
  }
});
console.log('Done fixing scss files!');
